src/psyclone/psyGen.py

```diff
@@ -3697,3 +3814,54 @@ def name(self):
 
     def apply(self):
         return None, None
+
+
+class IfBlock(Node):
+    '''
+    Class representing an if-block within a Schedule.
+    '''
+    def __init__(self, parent=None):
+        super(IfBlock, self).__init__(parent=parent)
+        self._condition = ""
+
+    def __str__(self):
+        return "If-block: "+self._condition
+
+    @property
+    def coloured_text(self):
+        '''
+        Return text containing the (coloured) name of this node type
+
+        :return: the name of this node type, possibly with control codes
+                 for colour
+        :rtype: string
```

Comment:
Isn't it str? And could you include this as part of the return seeing as it' s a simple type?

Reply:
It is (str) and Stackoverflow says I can't :-( (http://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#info-field-lists). Updated.

Reply:
Oops, rtfm. Although it would be nicer.